sormas-ui/src/main/java/de/symeda/sormas/ui/utils/BulkOperationHandler.java

```diff
@@ -38,78 +41,158 @@ public class BulkOperationHandler<T extends HasUuid> {
 	private int successfulEntryCount;
+	private Window window;
 
 	private final String allEntriesProcessedMessageProperty;
+	private final String ineligibleEntriesNotProcessedMessageProperty;
+	private final String headingSomeEntitiesNotDeleted;
+	private final String countEntriesNotDeletedMessageProperty;
 	private final String someEntriesProcessedMessageProperty;
+	private final String noEligibleEntityMessageProperty;
+	private final String infoBulkProcessFinishedWithSkipsProperty;
 
-	public BulkOperationHandler(String allEntriesProcessedMessageProperty, String someEntriesProcessedMessageProperty) {
+	public BulkOperationHandler(
+		String allEntriesProcessedMessageProperty,
+		String ineligibleEntriesNotProcessedMessageProperty,
+		String headingSomeEntitiesNotDeleted,
+		String countEntriesNotProcessedMessageProperty,
+		String someEntriesProcessedMessageProperty,
+		String noEligibleEntityMessageProperty,
+		String infoBulkProcessFinishedWithSkipsProperty) {
 		this.allEntriesProcessedMessageProperty = allEntriesProcessedMessageProperty;
+		this.ineligibleEntriesNotProcessedMessageProperty = ineligibleEntriesNotProcessedMessageProperty;
+		this.headingSomeEntitiesNotDeleted = headingSomeEntitiesNotDeleted;
+		this.countEntriesNotDeletedMessageProperty = countEntriesNotProcessedMessageProperty;
 		this.someEntriesProcessedMessageProperty = someEntriesProcessedMessageProperty;
+		this.noEligibleEntityMessageProperty = noEligibleEntityMessageProperty;
+		this.infoBulkProcessFinishedWithSkipsProperty = infoBulkProcessFinishedWithSkipsProperty;
 	}
 
 	public static <E extends HasUuid> BulkOperationHandler<E> forBulkEdit() {
-		return new BulkOperationHandler<E>(Strings.messageEntriesEdited, Strings.messageEntriesEditedExceptArchived);
+		return new BulkOperationHandler<E>(
+			Strings.messageEntriesEdited,
+			null,
+			null,
+			null,
+			Strings.messageEntriesEditedExceptArchived,
+			null,
+			Strings.infoBulkProcessFinishedWithSkips);
 	}
 
 	public void doBulkOperation(
 		Function<List<T>, Integer> bulkOperationFunction,
 		List<T> selectedEntries,
+		List<T> selectedEligibleEntries,
+		List<T> selectedIneligibleEntries,
 		Consumer<List<T>> bulkOperationDoneCallback) {
 
 		initialEntryCount = selectedEntries.size();
+		initialEligibleEntryCount = getInitialEligibleEntryCount(selectedEntries, selectedIneligibleEntries, selectedEligibleEntries);
+		selectedEligibleEntries = !areIneligibleEntriesSelected(selectedIneligibleEntries) ? selectedEntries : selectedEligibleEntries;
+
 		if (selectedEntries.size() < BULK_ACTION_PROGRESS_THRESHOLD) {
-			successfulEntryCount = bulkOperationFunction.apply(selectedEntries);
-			if (initialEntryCount > successfulEntryCount) {
+			if (initialEligibleEntryCount > 0) {
+				if (areIneligibleEntriesSelected(selectedIneligibleEntries)) {
+					successfulEntryCount =
+						areIneligibleEntriesSelected(selectedIneligibleEntries) ? bulkOperationFunction.apply(selectedEligibleEntries) : 0;
```

Comment:
Calling `areIneligibleEntriesSelected` seems to be redundant here because it's already done in the if condition